server/src/main/java/org/opensearch/index/remote/RemoteStoreEnums.java

```diff
@@ -92,14 +92,17 @@ public String getName() {
     public enum PathType {
         FIXED(0) {
             @Override
-            public BlobPath generatePath(PathInput pathInput, PathHashAlgorithm hashAlgorithm) {
+            public BlobPath generatePath(RemoteStorePathStrategy.BasePathInput pathInput, PathHashAlgorithm hashAlgorithm) {
                 assert Objects.isNull(hashAlgorithm) : "hashAlgorithm is expected to be null with fixed remote store path type";
                 // Hash algorithm is not used in FIXED path type
-                return pathInput.basePath()
-                    .add(pathInput.indexUUID())
-                    .add(pathInput.shardId())
-                    .add(pathInput.dataCategory().getName())
-                    .add(pathInput.dataType().getName());
+                BlobPath path = pathInput.basePath().add(pathInput.indexUUID());
+                if (pathInput instanceof RemoteStorePathInput) {
+                    RemoteStorePathInput remoteStorePathInput = (RemoteStorePathInput) pathInput;
+                    path.add(remoteStorePathInput.shardId())
+                        .add(remoteStorePathInput.dataCategory().getName())
+                        .add(remoteStorePathInput.dataType().getName());
+                }
+                return path;
             }
```

Comment:
The if/else conditions looks unnecessary and can make the code unwieldy. I would suggest we model `PathInput`  in such a way that we can have optional inputs.

Reply:
Since PathInput is marked as PublicApi, making any changes to interface to make inputs optional will be difficult. I feel creating a new base class handles BWC.